agents-and-function-calling/bedrock-agents/features-examples/11-create-agent-using-CDK-with-knowledge-base-and-action-group/bedrockagent/bedrockagent_stack.py

```diff
@@ -0,0 +1,446 @@
+        s3Bucket.add_event_notification(s3.EventType.OBJECT_REMOVED,
+                                      s3n.LambdaDestination(kb_sync_lambda))
+
+        # Add an explicit dependency on the lambda, so that the bucket
+        # deployment is started after the lambda is in place
+    
+        bucket_deployment.node.add_dependency(kb_sync_lambda)
+         
+        # Create the DynamoDB table
+        dynamodbable=dynamodb.Table(self, "restaurant_bookings",
+            partition_key=dynamodb.Attribute(
+                name="booking_id",
+                type=dynamodb.AttributeType.STRING
+            ),
+            table_name=table_name,
+            billing_mode=dynamodb.BillingMode.PAY_PER_REQUEST,
+            removal_policy=RemovalPolicy.DESTROY
+            )
+        
+        agent_role.add_to_policy(iam.PolicyStatement(sid='InvokeBedrockLambda',
+                                                     effect=iam.Effect.ALLOW,
+                                                     resources=[agent_model_arn],
+                                                     actions=['bedrock:InvokeModel','lambda:InvokeFunction']))
+        agent_role.add_to_policy(iam.PolicyStatement(sid='RetrieveKBStatement',
+                                                     effect=iam.Effect.ALLOW,
+                                                     resources=[knowledge_base.attr_knowledge_base_arn],
+                                                     actions=['bedrock:Retrieve']))
+        # Create the lambda function for the Agent Action Group
+        action_group_function = lambda_.Function(self, "BedrockAgentActionGroupExecutor",
+                    runtime=lambda_.Runtime.PYTHON_3_12,
+                    code=lambda_.Code.from_asset('lambda/actiongroup'),
+                    handler='lambda_function.lambda_handler',
+                    timeout=Duration.seconds(300)
+        )
+        action_group_function.add_to_role_policy(iam.PolicyStatement(sid="UpdateDynamoDB",
+                                                     effect=iam.Effect.ALLOW,
+                                                     resources=[dynamodbable.table_arn],
+                                                     actions=['dynamodb:GetItem','dynamodb:PutItem','dynamodb:DeleteItem']))
+                
+        cfn_agent = bedrock.CfnAgent(self, "CfnAgent",
+            agent_name= agent_name,
+
+            # the properties below are optional
+            action_groups=[bedrock.CfnAgent.AgentActionGroupProperty(
+                action_group_name= agent_action_group_name,
+                description=agent_action_group_description,
+
+                # the properties below are optional
+                action_group_executor=bedrock.CfnAgent.ActionGroupExecutorProperty(
+                    lambda_= action_group_function.function_arn
+                ),
+                                
+                function_schema=bedrock.CfnAgent.FunctionSchemaProperty(
```

Comment:
This needs to be more general... We should be able to get those from parameters in a configuration file and it should be expandable for different types of parameters/different numbers of parameters